File: c/examples/linked_list_example.c
#include <stdio.h>
#include <stdlib.h>
#include "ds/linked_list.h"

int main(void){
    LinkedList *list = createLinkedList();

    int *a = malloc(sizeof(int));
    int *b = malloc(sizeof(int));

    *a = 10;
    *b = 20;

    pushBack(list, a);
    pushBack(list, b);

    printf("Size: %zu\n", getSize(list));

    int *x = popFront(list);
    printf("Pop: %d\n", *x);
    free(x);

    destroyLinkedList(list, free);
    
    return 0;
}

File: python/examples/linked_list_example.py
from python.algods.ds.my_linked_list import LinkedList

def menu():
    l1 = LinkedList()

    while(True):
        print('SINGLY LINKED LIST MEU:')
        print('1: Push front')
        print('2: Push back')
        print('3: Pop front')
        print('4: Get size')
        print('5: Print')
        print('6: Exit')

        option = input("Select an option: ")

        if (option == '1'):
            val = input("Insert data: ")
            l1.push_front(val)
        elif (option == '2'):
            val = input("Insert data: ")
            l1.push_back(val)
        elif (option == '3'):
            l1.pop_front()
        elif (option == '4'):
            l1.get_size()
        elif (option == '5'):
            l1.print_list()
        else:
            print('Insert a correct number')

if __name__ == '__main__':
    menu()